### PR TITLE
Use Casters instead of Presenters as of PsySH ~0.5

### PR DESCRIPTION
**Pull request by nicolas-grekas**
PsySH ~0.5 [is replacing the presenter system by symfony/var-dumper](https://github.com/bobthecow/psysh/pull/184), so ezsh needs an update, here it is :)
Sorry, I didn't test this PR, but still, this should do the work. Please tell me if I did anything wrong.

### CODE REVIEW

**Comment by lolautruche** [comment]
Thanks @nicolas-grekas !
I'll give it a try :)

File: composer.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
         }
     ],
     "require": {
-        "psy/psysh": "~0.2",
+        "psy/psysh": "~0.5",
```
Comment (lolautruche) [inline composer.json]: Please specify `~0.5@dev` for now, as it has not been released yet.

File: src/EzCaster.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        $prefix = Caster::PREFIX_PROTECTED;
+
+        if (!isset($a[$prefix.'id'])) {
+            $b = (array) $location;
+            if (isset($b[$prefix.'id'])) {
+                $a += [$prefix.'id' => $b[$prefix.'id']];
+            }
+        }
+
+        if ($filter & Caster::FILTER_VERBOSE) {
```
Comment (lolautruche) [inline src/EzCaster.php]: This causes a fatal: 

```
Fatal error: Undefined class constant 'FILTER_VERBOSE' in /Users/lolautruche/workspace/ezsystems/ezsh/src/EzCaster.php on line 84
```

And indeed, there is no such constant in `Symfony\Component\VarDumper\Caster\Caster`

File: src/EzCaster.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    public function castContentInfo(ContentInfo $info, array $a, Stub $stub, $isNested, $filter = 0)
+    {
+        $a = [
+            Caster::PREFIX_PROTECTED.'id' => $info->id,
+            Caster::PREFIX_PROTECTED.'name' => $info->name,
+        ] + $a;
+
+        if (!$filter & Caster::FILTER_VERBOSE) {
```
Comment (nicolas-grekas) [inline src/EzCaster.php]: Sorry, it's Caster::EXCLUDE_VERBOSE, updated

File: composer.json (hunk 1 above)
Comment (bobthecow) [inline composer.json]: @lolautruche Specify it as `~0.5` :)

File: composer.json [hunk 4]
```diff
@@ -10,7 +10,7 @@
         }
     ],
     "require": {
-        "psy/psysh": "~0.2",
+        "psy/psysh": "~0.5@dev",
```
Comment (lolautruche) [inline composer.json]: Now that `0.5` is released, `@dev` is not needed any more.

**Comment by lolautruche** [comment]
Many thanks @nicolas-grekas !

I'll fix the dependency number.

**Pull request merged**